### PR TITLE
Add dependency security check using trivy

### LINKED ISSUE #899: Add tool to check vulnerabilities in go dependencies
- https://github.com/sonatype-nexus-community/nancy
- ?

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #899

This could be also done for docker image security check but that would probably require building docker image to tar file so I did not add that in this PR.

I usually do this with such steps:

```yaml
  docker-dryrun:
    image: thegeeklab/drone-docker-buildx:20
    settings:
      registry: xxx
      username:
        from_secret: docker_username
      password:
        from_secret: docker_password
      repo: xxx/${CI_REPO_OWNER}/${CI_REPO_NAME}
      tags: test
      dry_run: true
      output: type=oci,dest=${CI_REPO_OWNER}-${CI_REPO_NAME}.tar
    when:
      event:
      - pull_request

  docker-image-security-check:
    image: aquasec/trivy:latest
    commands:
      - trivy image --exit-code 0 --severity UNKNOWN,LOW --input ${CI_REPO_OWNER}-${CI_REPO_NAME}.tar
      - trivy image --exit-code 1 --severity MEDIUM,HIGH,CRITICAL --input ${CI_REPO_OWNER}-${CI_REPO_NAME}.tar
    when:
      event:
        - pull_request
```

### CODE REVIEW

File: .woodpecker/test.yml [hunk 1]
```diff
@@ -50,6 +50,15 @@ pipeline:
     image: mstruebing/editorconfig-checker
     group: test
 
+  securitycheck:
+    group: test
+    image: aquasec/trivy:latest
+    commands:
+      - trivy fs --exit-code 0 --skip-dirs web/ --severity UNKNOWN,LOW .
```
Comment (6543) [inline .woodpecker/test.yml:57]: we sould skip docs too and add a step for it to docs pipeline

Comment (lafriks) [reply .woodpecker/test.yml]: done

File: .woodpecker/docs.yml [hunk 2]
```diff
@@ -15,6 +15,14 @@ pipeline:
       event: [push, pull_request]
       path: *when_path
 
+  securitycheck:
+    image: aquasec/trivy:latest
+    commands:
+      - trivy fs --exit-code 0 --skip-dirs docs/node_modules --skip-dirs docs/plugins/woodpecker-plugins/node_modules --severity UNKNOWN,LOW docs/
```
Comment (6543) [inline .woodpecker/docs.yml:21]:
```suggestion
      # TODO: solve issue at docs/plugins/woodpecker-plugins and remove skip
      - trivy fs --exit-code 0 --skip-dirs docs/node_modules --skip-dirs docs/plugins/woodpecker-plugins/node_modules --severity UNKNOWN,LOW docs/
```

File: .woodpecker/docs.yml [hunk 3]
```diff
@@ -15,6 +15,15 @@ pipeline:
       event: [push, pull_request]
       path: *when_path
 
+  securitycheck:
+    image: aquasec/trivy:latest
+    commands:
+      # TODO: solve issue at docs/plugins/woodpecker-plugins and remove skip
```
Comment (6543) [inline .woodpecker/docs.yml:21]:
```suggestion
```

File: .woodpecker/docs.yml (hunk 2 above)
Comment (6543) [inline .woodpecker/docs.yml:21]: ah we only check against package.*

**Review by 6543 — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
well before we can merge it, I guess we have to bump frontend deps ...

... witch we also should do for release branch as long as possible ...

**Comment by lafriks** [comment]
should I fix security issues here or on other PR?

**Comment by 6543** [comment]
I dont mind ...

**Comment by lafriks** [comment]
I changed not to fail security check step for docs as it does not seems to be fixable at the moment 😕

**Comment by 6543** [comment]
-> `https://github.com/aquasecurity/trivy/pull/2823`

**Pull request merged**